### PR TITLE
rename module from akka to pekko

### PR DESCRIPTION
**Pull request by pjfanning**
WIP
* need to fix some test issues
* have updated some docs to use Pekko name
  * I'm reluctant to update the docs too much because there are a lot of links to docs.akka.io and we have no equivalent
  * security and migration guides are very Akka specific and probably need serious pruning

### CODE REVIEW

File: docs/src/main/paradox/compatibility-guidelines.md [hunk 1]
```diff
@@ -113,25 +113,25 @@ Java
...
-of your libraries: `Detected java.lang.NoSuchMethodError error, which MAY be caused by incompatible Akka versions on the classpath. Please note that a given Akka version MUST be the same across all modules of Akka that you are using, e.g. if you use akka-actor [2.5.3 (resolved from current classpath)] all other core Akka modules MUST be of the same version. External projects like Alpakka, Persistence plugins or Akka HTTP etc. have their own version numbers - please make sure you're using a compatible set of libraries.`
+of your libraries: `Detected java.lang.NoSuchMethodError error, which MAY be caused by incompatible Akka versions on the classpath. Please note that a given Akka version MUST be the same across all modules of Akka that you are using, e.g. if you use akka-actor [2.5.3 (resolved from current classpath)] all other core Apache Pekko modules MUST be of the same version. External projects like Alpakka, Persistence plugins or Apache Pekko HTTP etc. have their own version numbers - please make sure you're using a compatible set of libraries.`
```
Comment (nvollmar) [inline docs/src/main/paradox/compatibility-guidelines.md:130]: Missed an `Akka` here

File: docs/src/main/paradox/compatibility-guidelines.md [hunk 2]
```diff
@@ -113,25 +113,25 @@ Java
...
 
 ### Compatibility with Akka
 
-Akka HTTP 10.2.x is (binary) compatible with Akka >= $akka.minimum.version25$ and Akka >= $akka.minimum.version26$ and future Akka 2.x versions that are released during the lifetime of Akka HTTP 10.2.x.
+Apache Pekko HTTP 10.2.x is (binary) compatible with Akka >= $akka.minimum.version25$ and Akka >= $akka.minimum.version26$ and future Akka 2.x versions that are released during the lifetime of Apache Pekko HTTP 10.2.x.
```
Comment (nvollmar) [inline docs/src/main/paradox/compatibility-guidelines.md:134]: Referencing akka versioning

**Review by nvollmar — APPROVED** [review]
lgtm

File: docs/src/main/paradox/compatibility-guidelines.md (hunk 1 above)
Comment (pjfanning) [reply docs/src/main/paradox/compatibility-guidelines.md]: thank - I think I've fixed up the issues you raised - I'll double cjeck later

there is also at least 1 test issue that I need to debug

**Comment by pjfanning** [comment]
I'm going to merge this since the build is passing

**Pull request merged**